packages/store-cache/src/utils.ts

```diff
@@ -0,0 +1,194 @@
+import { ScanArgs, Tuple, TupleDatabaseClient } from "tuple-database";
+import { AbiDefaults } from "./defaults";
+import { StoreConfig } from "@latticexyz/store";
+import {
+  Key,
+  RemoveOptions,
+  SetOptions,
+  SubscriptionCallback,
+  SubscriptionFilterOptions,
+  Update,
+  Value,
+} from "./types";
+
+/**
+ * Set the value for the given key
+ * @param client TupleDatabaseClient
+ * @param table Table to set the given key in
+ * @param key Key to identify the record to set in the table
+ * @param value Value to set for the given key
+ * @param options {
+ *  appendToTransaction?: Append to an existing transaction, do not commit
+ *  defaultValue?: Default value to use if the key does not exist
+ * }
+ * @returns Transaction
+ */
+export function set<C extends StoreConfig = StoreConfig, T extends keyof C["tables"] = keyof C["tables"]>(
+  client: TupleDatabaseClient,
+  table: T,
+  key: Key<C, T>,
+  value: Partial<Value<C, T>>,
+  options?: SetOptions
+) {
+  const keyTuple = [String(table), ...recordToTuple(key)];
```

Comment:
> why do we need to convert table to string here? I would have expected keyof C['tables'] to be a string already?

Just to maks ts happy (it thought `table` can be `string | number | symbol`), but instead of casting here I just made the `table` type stronger now